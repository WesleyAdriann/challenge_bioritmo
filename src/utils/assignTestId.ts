type componentType = 'div' | 'text' | 'input' | 'button' | 'img'

export const assignTestId = (
  componentType: componentType,
  componentId: string
)  => ({
    'data-testid': `${componentType}_${componentId}`,
  })


export const mockTestID = (
    componentType: componentType,
    componentId: string
  ) => assignTestId(componentType, componentId)
