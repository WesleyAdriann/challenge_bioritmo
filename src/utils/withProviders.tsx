
import React from 'react'

import { ThemeProvider } from 'styled-components'

import { theme } from '../style'

export const StorybookProviders: React.FC = ({
  children,
}) => (
  <ThemeProvider theme={theme}>
    {children}
  </ThemeProvider>
)
