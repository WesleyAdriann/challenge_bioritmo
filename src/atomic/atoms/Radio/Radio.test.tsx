import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'

import { mockTestID, StorybookProviders as wrapper } from '../../../utils'

import { Radio, IRadio } from './index'

describe('Atoms/Radio', () => {
  const props: IRadio = {
    testID: 'Text',
    active: true,
    onClick: jest.fn()
  }

  const renderComponent = () => render(<Radio {...props} />, { wrapper })

  let component = renderComponent()

  beforeEach(() => {
    jest.clearAllMocks()
    cleanup()
    component = renderComponent()
  })

  const el_container = mockTestID('div', props.testID!)

  it(`should render component #${el_container}`, async () => {
    const sut = await component.findByTestId(el_container)
    expect(sut).toBeTruthy()
  })

  it('should render dot icon in Radio with active true', async () => {
    component.rerender(<Radio {...props} />)
    const sut = await component.findByTestId(el_container)
    expect(sut.children.length).toBeTruthy()
  })

  it('should not render dot icon in Radio with active false', async () => {
    component.rerender(<Radio {...props} active={false} />)
    const sut = await component.findByTestId(el_container)
    expect(sut.children.length).toBeFalsy()
  })

  it('should call onClick prop on click in Radio', async () => {
    const sut = await component.findByTestId(el_container)
    fireEvent.click(sut)
    expect(props.onClick).toBeCalled()
  })

  it('should render snapshot', () => {
    expect(component.container.firstChild).toMatchSnapshot()
  })
})
