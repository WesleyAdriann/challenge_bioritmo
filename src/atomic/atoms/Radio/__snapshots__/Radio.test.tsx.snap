// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/Radio should render snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFB612;
  border: 3px solid #808080;
  border-radius: 50%;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 24px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 2px;
  width: 24px;
}

.c0:active {
  -webkit-filter: brightness(105%);
  filter: brightness(105%);
}

.c1 {
  background-color: #000;
  border-radius: 50%;
  height: 12px;
  width: 12px;
}

<div
  class="c0"
  data-testid="div_Text"
  size="24"
>
  <div
    class="c1"
    size="24"
  />
</div>
`;
